### PR TITLE
Move px4 image to emmc

### PR DESCRIPTION
**Pull request by jlaitine**
WIP

### CODE REVIEW

File: platforms/nuttx/src/bootloader/microchip/mpfs/main.c [hunk 1]
```diff
@@ -263,15 +310,24 @@ board_deinit(void)
 	//	putreg32(RCC_AHB1RSTR_OTGFSRST, STM32_RCC_AHB1RSTR);
 #endif
 
-#ifdef CONFIG_MTD_M25P
-	mpfs_spibus_uninitialize(spinor);
-#endif
+#if CONFIG_MMCSD
```
Comment (pussuw) [inline platforms/nuttx/src/bootloader/microchip/mpfs/main.c:313]: #ifdef CONFIG_MMCSD

File: platforms/nuttx/src/bootloader/microchip/mpfs/main.c [hunk 2]
```diff
@@ -372,8 +459,20 @@ flash_func_erase_sector(unsigned sector)
...
+	/* Erase the whole file when erasing the first sector */
+	if (sector == 0) {
+		create_px4_file();
+	}
+
+#endif
+
+	/* Erase the RAM contents */
+
 	memset(addr, 0xFFFFFFFF, ss);
```
Comment (pussuw) [inline platforms/nuttx/src/bootloader/microchip/mpfs/main.c:474]: Is there some not obvious reason from memsetting RAM to all 1's ? Seems like some funny remnant of flash erase.

File: platforms/nuttx/src/bootloader/microchip/mpfs/main.c [hunk 3]
```diff
@@ -590,81 +698,32 @@ static size_t get_image_size(void)
...
-	parse_block_partition("/dev/mmcsd0", partition_handler, "/sdcard/");
-	_alert("%s %d\n", __func__, __LINE__);
-	/*
-	 * Mount the sdcard and check if the image is present
-	 */
-	int ret = mount("/dev/mmcsd0p0", "/sdcard/", "vfat", 0, NULL);
-
-	if (ret >= 0) {
-		int mmc_fd = open("/sdcard/boot/" IMAGE_FN, O_RDWR | O_CREAT);
+	//_alert("%s %d\n", __func__, __LINE__);
```
Comment (pussuw) [inline platforms/nuttx/src/bootloader/microchip/mpfs/main.c:711]: Personally I would prefer leaving commented code out, or at least give a reason why the code is there and why it is commented.

The informative value of this print is not immediately obvious to me, seems like something that was used/added for debugging.

File: platforms/nuttx/src/bootloader/microchip/mpfs/main.c [hunk 4]
```diff
@@ -337,11 +393,40 @@ flash_func_sector_size(unsigned sector)
...
+		if (ret < 0) {
+			_alert("boot directory creation failed %d\n", ret);
+		}
+
+		px4_fd = open("/sdcard/boot/" IMAGE_FN, O_RDWR | O_CREAT);
+	}
+
+	if (px4_fd < 0) {
+		_alert("FATAL: Not able to create px4 fw image!\n");
+
```
[the remark this answers is not in the document]
Comment (jlaitine) [reply platforms/nuttx/src/bootloader/microchip/mpfs/main.c:415]: Yes, I guess it was something automatically added by "make format"

File: platforms/nuttx/src/bootloader/microchip/mpfs/main.c (hunk 2 above)
Comment (jlaitine) [reply platforms/nuttx/src/bootloader/microchip/mpfs/main.c:474]: Yes ;) in common code it "verifies" that the erase succeeded by reading back and checking that it is ffff. Doesn't make much sense with RAM but keeps that part happy...

File: platforms/nuttx/src/bootloader/microchip/mpfs/main.c (hunk 3 above)
Comment (jlaitine) [reply platforms/nuttx/src/bootloader/microchip/mpfs/main.c:711]: Yep, leftovers from some debug. Will remove

File: platforms/nuttx/src/bootloader/microchip/mpfs/main.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (jlaitine) [reply platforms/nuttx/src/bootloader/microchip/mpfs/main.c:414]: Verified this, astyle just adds it

File: platforms/nuttx/src/bootloader/microchip/mpfs/main.c (hunk 3 above)
Comment (jlaitine) [reply platforms/nuttx/src/bootloader/microchip/mpfs/main.c]: Now removed

**Review by pussuw — APPROVED** [review]
Ty for the answers! LGTM

**Comment by jlaitine** [comment]
I still updated nuttx submodule to incorporate one eMMC patch from @eenurkka

**Pull request merged**